zokrates_core/src/standard.rs

```diff
@@ -37,6 +37,11 @@ pub struct Constraint {
 	c: BTreeMap<usize, String>,
 }
 
+pub struct DirectiveR1CS {
+    pub r1cs : R1CS,
+    pub directive : Option<LibsnarkGadgetHelper>
```

Comment:
Is there any case where we wouldn't want a directive? Wouldn't it end being a bunch of constraints the wires of which are not assigned?

Reply:
Correct me if I'm wrong, but I believe the directive is only necessary when we have a computation done off-circuit, which is then verified on-circuit? So it would be the case that for some gadgets this is not necessary.

Reply:
I guess @Schaeff is right. `R1CS` is only being used if libsnark is involved, hence there also will be usage of some directive / helper.
`None` is currently used for example in the following test:
https://github.com/JacobEberhardt/ZoKrates/pull/136/files#diff-e4af2ccde827780269c4d04b27eef2ccR262
I guess this should be changed as well.

My main reasoning was to only have on conversion into `FlatFunction`.

Reply:
Yep @sanchopansa as this yields Constraints, the variables involved need to have been defined before either in a Definition or a Directive, so I can't see the case for not having a Directive here. Adjusting the test by specifying the sha256 directive instead of None would work, and hopefully we can keep directives and constraints at the same place in the future

Reply:
Changed as discussed.